github/github.go

```diff
@@ -739,7 +739,7 @@ func (c *Client) RateLimit() (*Rate, *Response, error) {
 		return nil, nil, err
 	}
 
-	return limits.Core, resp, err
+	return limits.Core, resp, nil
```

Comment:
Hmm, this one won't work, it changes behavior. But I think the problem lies above.

Currently, it checks if limits is nil, and always returns err. In the (possibly impossible) path where limits is not nil, but err is also not nil, this change would incorrectly return nil error.

However, I think you should change the checks above to be like this:

```Go
limits, resp, err := c.RateLimits()
if err != nil {
	return nil, nil, err
}
if limits == nil {
	return nil, nil, fmt.Errorf("RateLimits() returned nil error and nil limits, cannot extract Core rate limit")
}
```

Then it's fine to do `return limits.Core, resp, nil` here.

We could also leave this as is, but I think it's unfriendly API to return `nil, nil, nil`, so changing it to always return non-nil error on failure would be better.